src/model/formats/format-registry.ts

```diff
@@ -0,0 +1,142 @@
+import { AceFormat } from './ace-format';
+import { JsonFormat } from './json-format';
+import { DynonFormat, DynonFormatOptions } from './dynon-format';
+import { GrtFormat } from './grt-format';
+import { ForeFlightFormat } from './foreflight-format';
+import { PdfWriterOptions } from './pdf-writer';
+import { PdfFormat } from './pdf-format';
+import { ChecklistFile } from '../../../gen/ts/checklist';
+import {
+  AbstractChecklistFormat,
+  DefaultFormatOptions,
+  FileExtension,
+  FormatConstructor,
+  FormatOptions,
+  getFileExtension,
+  OutputFormat,
+} from './abstract-format';
+import { FormatError } from './error';
+import { TXT_EXTENSION } from './text-format-options';
+
+declare global {
+  const enum FormatId {
+    ACE = 'ace',
+    AFD = 'afd',
+    DYNON = 'dynon',
+    DYNON31 = 'dynon31',
+    DYNON40 = 'dynon40',
+    FOREFLIGHT = 'fmd',
+    GRT = 'grt',
+    JSON = 'json',
+    PDF = 'pdf',
+  }
+}
+
+class FormatRegistry {
+  private readonly _outputFormats = new Map<FormatId, AbstractChecklistFormat>();
+  private readonly _inputFormats = new Map<FileExtension, AbstractChecklistFormat[]>();
+
+  public register<T extends FormatOptions>(
+    formatId: FormatId,
+    constructor: FormatConstructor<T>,
+    formatOptions: T,
+  ): void {
+    if (this._outputFormats.has(formatId)) {
+      throw new Error(`Format "${formatId}" already registered`);
+    }
+
+    const format = new constructor(formatId, formatOptions);
+    this._outputFormats.set(formatId, format);
+
+    if (format.supportsInput) {
+      const registeredFormats = this._inputFormats.get(format.extension);
+      if (registeredFormats !== undefined) {
+        registeredFormats.push(format);
+      } else {
+        this._inputFormats.set(format.extension, [format]);
+      }
+    }
+  }
+
+  public getFormat(formatId: FormatId): AbstractChecklistFormat {
+    const format = this._outputFormats.get(formatId);
+    if (format === undefined) {
+      throw new Error(`Format '${formatId}' not registered`);
+    }
+    return format;
+  }
+
+  public getFormatsByExtension(extension: string): AbstractChecklistFormat[] {
```

Comment:
`extension: FileExtension` ?

Reply:
Implemented in 9fae08a6f0898091639311cc528c80972555f9a5, thanks!